Source/Diagnostics/Diagnostics.cpp

```diff
@@ -117,8 +117,8 @@ Diagnostics::BaseReadParameters ()
     }
 #ifdef WARPX_DIM_RZ
     WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
-        m_pfield_varnames.empty(),
-        "Input error: cannot use particle_fields_to_plot with RZ"
+        (WarpX::ncomps == 1),
```

Comment:
Would this diagnostic work if there are more than one mode, but have the diagnostic only write out the mode 0 values? If so, this check would not be needed.

Reply:
The way this is currently implemented this would be true. 

I'm not sure how mode decomposed output data looks like actually, so it may not be obvious that it is the 0th mode only.